sycl/include/sycl/detail/pi.h

```diff
@@ -1790,6 +1800,62 @@ __SYCL_EXPORT pi_result piextUSMGetMemAllocInfo(
     pi_context context, const void *ptr, pi_mem_alloc_info param_name,
     size_t param_value_size, void *param_value, size_t *param_value_size_ret);
 
+/// USM 2D fill API
+///
+/// \param queue is the queue to submit to
+/// \param ptr is the ptr to fill
+/// \param pitch is the total width of the destination memory including padding
+/// \param pattern is a pointer with the bytes of the pattern to set
+/// \param pattern_size is the size in bytes of the pattern
+/// \param width is width in bytes of each row to fill
+/// \param height is height the columns to fill
+/// \param num_events_in_waitlist is the number of events to wait on
+/// \param events_waitlist is an array of events to wait on
+/// \param event is the event that represents this operation
+__SYCL_EXPORT pi_result piextUSMEnqueueFill2D(pi_queue queue, void *ptr,
```

Comment:
The similar "buffer" API use "Rect" instead of "2D". I think we should unify names if possible (especially for Unified Runtime). Tag @kbenzie

Reply:
@veselypeta this is relevant to https://github.com/oneapi-src/unified-runtime/pull/107

Reply:
@smaslov-intel the difference with `*Rect` though is they support 2D or 3D where as this is only 2D.

Reply:
Does it not make sense to add 3D for USM?

Reply:
if we were to split 2D/3D then maybe we should do the same to the existing Rect API four buffers
OK, this discussion belongs to Unified Runtime issue, I guess